earth_enterprise/src/fusion/autoingest/sysman/AssetHandleD.h

```diff
@@ -189,7 +189,13 @@ class MutableAssetHandleD_ : public virtual Base_ {
       for (std::vector<std::string>::iterator it = toDelete.begin();
            it != toDelete.end(); ++it) {
         if (dirtyMap.find(*it) != dirtyMap.end()) {
-          std::string filename = dirtyMap[*it]->XMLFilename() + ".new";
+          std::string filename = dirtyMap[*it]->XMLFilename();
```

Comment:
I'm not sure if this will completely fix the problem. I think we want to avoid getting files that end in .new into the dirtyMap in the first place. This will accomplish the goal of preventing .new.new files from appearing, but the underlying issue that .new files are being processed as if they were .xml files will still exist.

Thoughts?

Reply:
Were we not wanting to overwrite the files that are currently there?

Reply:
This will probably create bugs.  You'll be writing an XML file that doesn't correspond to a valid build of an asset.  E.g., until an asset build succeeds, there will be files files that are partially built, or metadata that has been updated, but the files it refers to won't be there, yet,or they won't have the correct contents.

If Fusion services are terminated before an asset build completes, you'll end up with an XML describing an invalid build that looks just like a successfully built asset.  If you kill Fusion services at the right time, it's probably also possible to end up with an incompletely written XML and loose data.

You also need to know that there's no other code that can request information about a building asset that has been purged from the cache.  Otherwise, that code will receive asset metadata that doesn't correspond to built files on disk.

If you've already read all the code, and know there are no such possible problems, it may be OK to overwrite existing XML files.  Otherwise, I think the first step is to add documentation about how asset matadata is updated during a build, and how the asset cache handles updates, especially during a cache purge.

Reply:
This gets back to a previous question I had about .new remnants being fully formed or not.

Reply:
I think we decided to back out the change to avoid creating a filename ending in .new.new. If I remember the conversation correctly, we thought it was safest to continue allowing .new.new files to be created and just log when they are being created.

Reply:
Note: you can still keep the notify statement for when we create a .new file.